src/Module/FollowConfirm.php

```diff
@@ -0,0 +1,93 @@
+<?php
+namespace Friendica\Module;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Core\L10n;
+use Friendica\Core\Logger;
+use Friendica\Core\Protocol;
+use Friendica\Database\DBA;
+use Friendica\Model\Contact;
+use Friendica\Protocol\Diaspora;
+use Friendica\Protocol\ActivityPub;
+use Friendica\Util\DateTimeFormat;
+
+class FollowConfirm extends BaseModule
+{
+	public static function post()
+	{
+		$a = self::getApp();
+
+		$uid = local_user();
+		if (!$uid) {
+			notice(L10n::t('Permission denied.') . EOL);
+			return;
+		}
+
+		$intro_id = intval(defaults($_POST, 'intro_id'  , 0));
+		$duplex   = intval(defaults($_POST, 'duplex'    , 0));
+		$cid      = intval(defaults($_POST, 'contact_id', 0));
+		$hidden   = intval(defaults($_POST, 'hidden'    , 0));
+
+		if (empty($cid)) {
+			notice(L10n::t('No given contact.') . EOL);
+			return;
+		}
+
+		Logger::log('Confirming follower with contact_id: ' . $cid);
+
+		$contact = DBA::selectFirst('contact', [], ['id' => $cid, 'uid' => $uid]);
+		if (!DBA::isResult($contact)) {
+			Logger::log('Contact not found in DB.');
+			notice(L10n::t('Contact not found.') . EOL);
```

Comment:
```suggestion
			throw new \Friendica\Network\HTTPException\NotFoundException(L10n::t('Contact not found.'));
```

Reply:
The notices are for cases where we don't want to throw hard errors.

Reply:
I concur, since it's in `post()` it would show a bare screen as well, which we probably don't want.